src/SMU-Keithley_6430/main.py

```
@@ -338,4 +505,28 @@ def read_measurement(self) -> float:
+        exception (sweep still running). This loop avoids blocking the whole program for the entire sweep
+        duration while still detecting a true instrument error if the elapsed time greatly exceeds the
+        expected sweep duration.
+        """
+        start_time = time.monotonic()
+        max_wait = self.list_expected_time * 3 + 30  # allow 3x expected time plus a fixed 30 s margin
+
+        while not self.is_run_stopped():
+            try:
+                self.port.query("*OPC?")
+                return  # instrument confirmed all operations complete
+            except Exception:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)